src/components/describe/describeWebview.ts

```diff
@@ -1,23 +1,69 @@
 import * as vscode from 'vscode';
+import { ShellResult } from '../../shell';
 import { WebPanel } from '../webpanel/webpanel';
 
 export class DescribePanel extends WebPanel {
     public static readonly viewType = 'vscodeKubernetesDescribe';
+    private static readonly describeContext = 'vscodeKubernetesDescribeContext';
+    private static activePanel: DescribePanel | undefined = undefined;
     public static currentPanels = new Map<string, DescribePanel>();
 
-    public static createOrShow(content: string, resource: string) {
+    public static refreshCommand() {
+        if (DescribePanel.activePanel) {
+            DescribePanel.activePanel.doRefresh();
+        }
+    }
+
+    public static createOrShow(content: string, resource: string, refresh: () => Promise<ShellResult | undefined>) {
         const fn = (panel: vscode.WebviewPanel, content: string, resource: string): DescribePanel => {
-            return new DescribePanel(panel, content, resource);
+            return new DescribePanel(panel, content, resource, refresh);
         };
         WebPanel.createOrShowInternal<DescribePanel>(content, resource, DescribePanel.viewType, "Kubernetes Describe", DescribePanel.currentPanels, fn);
     }
 
     private constructor(
         panel: vscode.WebviewPanel,
         content: string,
-        resource: string
+        resource: string,
+        readonly refresh: () => Promise<ShellResult | undefined>
     ) {
         super(panel, content, resource, DescribePanel.currentPanels);
+
+        const setActiveContext = (active: boolean) => {
+            vscode.commands.executeCommand('setContext', DescribePanel.describeContext, active);
+            DescribePanel.activePanel = active ? this : undefined;
+        };
+
+        this.panel.onDidChangeViewState((evt: vscode.WebviewPanelOnDidChangeViewStateEvent) => {
+            setActiveContext(evt.webviewPanel.active);
+        });
+        setActiveContext(true);
+        this.panel.webview.onDidReceiveMessage(
+            async (message) => {
+                switch (message.command) {
+                    case 'refresh':
+                        await this.doRefresh();
+                        break;
+                }
+            },
+            undefined
+        );
+    }
+
+    protected async doRefresh() {
```

Comment:
Why is this `protected`?  Looks like it could be `private`, or does that stop it being called by the static method somehow?

Reply:
made private.